app/controllers/secret_shares_controller.rb

```diff
@@ -0,0 +1,9 @@
+class SecretSharesController < ApplicationController
+
+  def create
+    moment = Moment.friendly.find(params[:moments])
+    moment.update :secret_share_identifier => SecureRandom.uuid
```

Comment:
Really, controllers should be delegators. While the distinction can get a little grey at times, mostly their work is asking different parts of the application to complete tasks and then packaging up the results for the response.

IMO, in this case, I would say that the controller shouldn't know that you create a share by generating a UUID which is stored in the secret_share_identifier of a moment. I think the controller would get the moment and then ask the moment to generate a share. Any logic regarding how that share is created is contained within the model.

I don't want to prescribe any particular naming conventions but just to help the discussion, it could be that the controller calls `moment.generate_secret_share!` which does the `update!` and returns true.

Take a look over [Active Record Validations](http://guides.rubyonrails.org/active_record_validations.html)—there is a section on [validating uniqueness](http://guides.rubyonrails.org/active_record_validations.html#uniqueness).

While the probability of 2 UUIDs clashing is extremely low, it is still possible.

As this solution develops, this validation may change to account for the expiry but you can start with a simple uniqueness validator and generate a test to prove that it works.